src/clients/evm.ts

```diff
@@ -161,18 +161,19 @@ const init = async (providerUrl: string): Promise<EVMClient> => {
       const owner = await contract.owner();
       return owner
     },
-    getBlockTimestampsByBlockNumber: async (blockNumbers: number[]) => {
+    getBlockTimestampsByBlockNumber: async (blockNumbers: number[]): Promise<{ [blockNumber: string]: string }> => {
       const getBlock = provider.getBlock.bind(provider);
       const results = await rollPromises(blockNumbers, getBlock);
```

Comment:
hmm so likely up to 1000 api calls to the endpoint? i think alchemy may throttle us but lets try and see. i guess with 120k+ transfers the first index could take quite long, but probably still fine if its first time only, especially if there are duplicates.

Reply:
It's been okay with my testing so far, but if we do run into trouble this will be easy to limit since we are already using rollPromises